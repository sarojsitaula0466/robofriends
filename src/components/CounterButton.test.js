import React from "react";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import { shallow, configure } from "enzyme";
import CounterButton from "./CounterButton";

configure({ adapter: new Adapter() });

it("expect to render CounterComponent component", () => {
  const mockColor = "red";
  const component = renderer
    .create(<CounterButton color={mockColor} />)
    .toJSON();
  expect(component).toMatchSnapshot();
});

it("correctly increments the counter", () => {
  const mockColor = "red";
  const wrapper = shallow(<CounterButton color={mockColor} />);
  wrapper.find("[id='counter']").simulate("click");
  wrapper.find("[id='counter']").simulate("click");
  expect(wrapper.state()).toEqual({ count: 3 });
});
