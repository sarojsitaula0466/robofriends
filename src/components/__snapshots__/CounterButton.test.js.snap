// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CounterComponent component 1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  Count: 
  1
</button>
`;
